### PR TITLE
[CI] Refactor existing workflows

### PR DESCRIPTION
**Pull request by alexbatashev**
Summary of changes:
- Remove some unused arguments
- Add more arguments to allow more LLVM Test Suite configurations
- Add ref arguments to allow calling workflow from another repository
- Switch to matrix builds for LLVM Test Suite: less verbose output by default and avoid code duplication
- Invoke main workflow from current commit rather than `sycl` branch
- Disable formatting for config files
- Minor renames to make workflows more consistent
- Uplift some dependencies

### CODE REVIEW

File: devops/scripts/generate_test_matrix.py [hunk 1]
```diff
@@ -0,0 +1,26 @@
+import json
+import sys
+import os
+
+with open(sys.argv[1]) as f:
+    config = json.load(f)
+    inputs = json.loads(os.environ['GHA_INPUT'])
+
+    enabled_lts_configs = inputs['lts_config'].slit(";")
```
Comment (alexbatashev) [inline devops/scripts/generate_test_matrix.py:9]:
```suggestion
    enabled_lts_configs = inputs['lts_config'].split(";")
```

File: devops/scripts/generate_test_matrix.js [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+
+      console.log(inputs);
+      console.log(ltsConfigs);
+
+      const enabledLTSConfigs = [];
+
+      testConfigs.lts.forEach(v => {
+        console.log(v);
+        if (ltsConfigs.includes(v.config)) {
+          console.log("BOO");
```
Comment (bader) [inline devops/scripts/generate_test_matrix.js:20]: What is "BOO"? Could add a comment for this line explaining the choice for this message?

**Review by bader — APPROVED** [review]

File: devops/scripts/generate_test_matrix.js (hunk 2 above)
Comment (alexbatashev) [reply devops/scripts/generate_test_matrix.js]: It's just a lack of proper debugger for these scripts. Removed debug output.

**Comment by bader** [comment]
> https://github.com/intel/llvm/runs/5167480274?check_suite_focus=true looks suspicious. Looks like a data race...

I see the same issue in post-commit as well  - https://github.com/intel/llvm/runs/5167562221?check_suite_focus=true

**Review by bader — APPROVED** [review]

**Pull request merged**